lib/commands/certificate.js

```diff
@@ -0,0 +1,188 @@
+    type: 'accessibility id',
+    value: 'Done'
+  });
+}
+
+/**
+ * @typedef {Object} CertificateInstallationOptions
+ *
+ * @property {!string} content - Base64-encoded content of the public certificate
+ */
+
+/**
+ * Installs a custom certificate onto the device.
+ * Since Apple provides no official way to do it via command line,
+ * this method tries to wrap the certificate into .mobileconfig format
+ * and then deploys the wrapped file to the internal HTTP server,
+ * so one can open it with mobile Safari.
+ * Then the algorithm goes through the profile installation procedure by
+ * clicking the necessary buttons using WebDriverAgent.
+ *
+ * @param {CertificateInstallationOptions} opts
+ * @returns {string} The content of the generated .mobileconfig file as
+ * base64-encoded string. This config might be useful for debugging purposes.
+ */
+commands.mobileInstallCertificate = async function (opts = {}) {
+  const {content} = opts;
+  if (_.isEmpty(content)) {
+    throw new Error('Certificate content should not be empty');
+  }
+
+  const staticContentRoot = path.resolve(
+    __dirname.substring(0, __dirname.lastIndexOf(SELF_MODULE_NAME)),
+    BASE_MODULE_NAME, 'static');
+  if (!await fs.exists(staticContentRoot)) {
+    throw new Error(`The static content root '${staticContentRoot}' ` +
+                    `does not exist or is not accessible`);
+  }
+  const configName = `${(Math.random() * 0x100000000 + 1).toString(36)}.${CONFIG_EXTENSION}`;
+  const configPath = path.resolve(staticContentRoot, configName);
+  const mobileConfig = await toMobileConfig(Buffer.from(content, 'base64'));
+  try {
+    await plist.updatePlistFile(configPath, mobileConfig, false, false);
+  } catch (err) {
+    throw new Error(`Cannot store the generated config as '${configPath}'. ` +
+                    `Original error: ${err.message}`);
+  }
+  try {
+    const {address, port} = this.server.address();
+    const certUrl = `http://${address ? address : os.hostname()}` +
+                    `:${port ? port : 4723}/${configName}`;
+    try {
+      if (this.isRealDevice()) {
+        try {
+          await this.proxyCommand('/url', 'POST', {url: certUrl});
+        } catch (err) {
+          if (this.isWebContext()) {
+            // The command above does not always work on real devices
+            await iosCommands.general.setUrl.call(this, certUrl);
+          } else {
+            throw err;
+          }
+        }
+      } else {
+        await openUrl(this.opts.udid || this.sim.udid, certUrl);
+      }
+
+      await installCertificateInPreferences(this);
+    } finally {
+      try {
+        await this.activateApp(this.opts.bundleId);
+      } catch (e) {
+        log.warn(`Cannot restore the application '${this.opts.bundleId}'. ` +
+                 `Original error: ${e.message}`);
+      }
+    }
+
+    return (await fs.readFile(configPath)).toString('base64');
+  } finally {
+    if (await fs.exists(configPath)) {
```

Comment:
No need to check. `await fs.rimraf(configPath);` will either delete the directory, if it exists, or quietly go on its way without doing anything, if it does not exist.